src/lib/common/include/sol-platform.h

```diff
@@ -156,6 +160,75 @@ int sol_platform_restart_service(const char *service);
 
 int sol_platform_set_target(const char *target);
 
+struct sol_mount_point {
+    char *dev;
+    char *mpoint;
+    char *fstype;
+    unsigned long flags;
+    char *opts;
+    bool noauto;
+};
```

Comment:
this shouldn't be exposed in a general header file like this, at least not the way it is.

flags, opts and dev are all linux-specific. One could argue that dev could be virtual and on other systems could be encoded as required (ie: "usb:xyz" for a device "xyz" in an usb bus in RIOT/Contiki, if these are ever supported). But should we bother to list where the device came from? Or the flags?

Right now I'd keep it simple and just do not expose anything but the mount point.